src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -201,26 +201,63 @@ public override bool CheckUsabilityOnScavenge()
             }
 
             // We may already have a read-ahead task if we did a previous scavenge and haven't used the connection since.
-#pragma warning disable CA2012 // we're very careful to ensure the ValueTask is only consumed once, even though it's stored into a field
-            _readAheadTask ??= ReadAheadWithZeroByteReadAsync();
-#pragma warning restore CA2012
+            EnsureReadAheadTaskHasStarted();
```

Comment:
Is it possible scavenges could end up overlapping such that two threads concurrently try to initialize the read-ahead? (Your PR wouldn't make that worse.)